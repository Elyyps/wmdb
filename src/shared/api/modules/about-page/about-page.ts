import { ILink } from "@app/api/core/link";

export interface IAboutPage {
  button: ILink;
  content: string;
  title: string;
}
