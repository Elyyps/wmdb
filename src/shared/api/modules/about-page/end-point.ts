import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
