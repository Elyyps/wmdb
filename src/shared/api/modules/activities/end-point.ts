import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
