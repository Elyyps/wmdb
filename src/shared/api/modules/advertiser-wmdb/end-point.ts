import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
