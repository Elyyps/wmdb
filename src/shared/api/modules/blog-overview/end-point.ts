import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
