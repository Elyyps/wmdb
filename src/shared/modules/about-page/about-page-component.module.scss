.about-page {
  &-button {
    margin-top: 24px;
  }
}
