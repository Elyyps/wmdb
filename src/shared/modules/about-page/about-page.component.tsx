import * as React from "react";
import styles from "./about-page-component.module.scss";
import { Section } from "@app/core/section";
import { BodyTextComponent } from "@app/core/bodytext";
import { IAboutPage } from "@app/api/modules/about-page/about-page";
import { Button } from "@app/core/button";

export interface IAboutPageComponentProps {
  aboutPage: IAboutPage;
}

const AboutPageComponent = (props: IAboutPageComponentProps) => (
  <Section title={props.aboutPage.title} paddingBottom={"40px"} paddingTop={"40px"} align="center">
    <div className={styles["about-page"]}>
      <BodyTextComponent html={props.aboutPage.content} />
      <div className={styles["about-page-button"]}>
        <Button title={props.aboutPage.button.title} variant="primary" href={props.aboutPage.button.url} />
      </div>
    </div>
  </Section>
);

export { AboutPageComponent };
