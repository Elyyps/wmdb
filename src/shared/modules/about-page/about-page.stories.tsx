import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { AboutPageComponent } from "./about-page.component";
import { generateAboutPageData } from "@app/api/modules/about-page/end-point";

storiesOf("AboutPage", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <AboutPageComponent aboutPage={generateAboutPageData()} />);
