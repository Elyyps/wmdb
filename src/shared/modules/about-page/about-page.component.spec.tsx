import React from "react";
import { shallow } from "enzyme";
import { AboutPageComponent } from "./about-page.component";
import { generateAboutPageData } from "@app/api/modules/about-page/end-point";

describe("[AboutPage]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<AboutPageComponent aboutPage={generateAboutPageData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
