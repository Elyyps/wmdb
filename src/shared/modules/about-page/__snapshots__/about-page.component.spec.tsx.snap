// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AboutPage] should render component without crashing 1`] = `
<Section
  align="center"
  paddingBottom="40px"
  paddingTop="40px"
  title="Over deze pagina"
>
  <div
    className="about-page"
  >
    <BodyTextComponent
      html="  
  <p>Er zijn heel veel citygames mogelijk in Amsterdam! U kunt zo de stad Amsterdam op een leuke manier beter leren
kennen. U kunt GPS tochten door de stad doen, moordtochten, puzzeltochten enz. U kunt lopend Amsterdam verkennen
tijdens uw citygame, maar ook per step, tuktuk, waterfiets of solex! Kiest u voor bepaalde wijken zoals de
Wallen, of gaat u liever Zuid.
</p>

<p>Nog niet alles duidelijk? Stel je vraag!</p>"
    />
    <div
      className="about-page-button"
    >
      <Button
        href="/"
        title="Stel je vraag"
        variant="primary"
      />
    </div>
  </div>
</Section>
`;
